### PR TITLE
Adding new machines specs

### PR DESCRIPTION
**Pull request by AleksandarCole**

### CODE REVIEW

File: docs/ci-cd-environment/machine-types.md [hunk 1]
```diff
@@ -69,21 +73,170 @@ Linux machine types can be paired with [Ubuntu 18.04 image][ubuntu1804], [Ubuntu
...
+  </td>
+</tr>
+</tbody>
+</table>
+
+Implementation details:
+
+1. `Virtual CPU` hyperthreaded on a 3.6GHZ [AMD Ryzen 5 3600](https://www.amd.com/en/product/8456).
+2. `Memory` is implemented as DDR4 RAM.
+3. `Disk` is implemented as RAM drive backed by DDR4 RAM.
```
Comment (bogyo210) [inline docs/ci-cd-environment/machine-types.md:162]: Disks are backed by NvME drives.

File: docs/ci-cd-environment/machine-types.md [hunk 2]
```diff
@@ -69,21 +73,170 @@ Linux machine types can be paired with [Ubuntu 18.04 image][ubuntu1804], [Ubuntu
...
+  </td>
+</tr>
+</tbody>
+</table>
+
+Implementation details:
+
+1. `Virtual CPU` hyperthreaded on a 4.6GHz 12th generation [Intel i5 125000](https://ark.intel.com/content/www/us/en/ark/products/96144/intel-core-i512500-processor-18m-cache-up-to-4-60-ghz.html).
+2. `Memory` is implemented as DDR4 RAM.
+3. `Disk` is implemented as RAM drive backed by DDR4 RAM.
```
Comment (bogyo210) [inline docs/ci-cd-environment/machine-types.md:237]: Disks are backed by NvME drives

**Review by bogyo210 — CHANGES_REQUESTED** [review]

File: docs/ci-cd-environment/machine-types.md [hunk 3]
```diff
@@ -15,6 +15,10 @@ the supported machine types in your pipelines, refer to our [agent documentation
 
 Linux machine types can be paired with [Ubuntu 18.04 image][ubuntu1804], [Ubuntu 20.04 image][ubuntu2004], and [Docker-based environment][docker-env].
 
+### E1 Generation
+
+The `e1` series machines are our first generation, cost-effective with medium performance they are a good choice for less compute and memory intensive jobs.
```
Comment (csidyel) [inline docs/ci-cd-environment/machine-types.md:20]: I'd split the sentence into two to keep them shorter.

The `e1` series machines are our first generation, cost-effective with medium performance. They are a good choice for less compute and memory-intensive jobs.

File: docs/ci-cd-environment/machine-types.md [hunk 4]
```diff
@@ -69,21 +73,170 @@ Linux machine types can be paired with [Ubuntu 18.04 image][ubuntu1804], [Ubuntu
...
+
+1. `Virtual CPU` hyperthreaded on a 3.6GHZ [AMD Ryzen 5 3600](https://www.amd.com/en/product/8456).
+2. `Memory` is implemented as DDR4 RAM.
+3. `Disk` is implemented as RAM drive backed by DDR4 RAM.
+
+`e2` machines are currently only available on our [enterprise plan](https://semaphoreci.com/pricing).
+
+### F1 Generation
+
+The `f1` series machines are built for compute-intensive jobs, these machines deliver highest performance..
```
Comment (csidyel) [inline docs/ci-cd-environment/machine-types.md:168]: deliver the highest performance.

there are two dots at the end of `performance..`

**Review by csidyel — CHANGES_REQUESTED** [review]
my notes

**Comment by AleksandarCole** [comment]
Closed in favor of: https://github.com/semaphoreci/docs/pull/1292
All suggestions and requested changes are implemented in that PR. Thanks @bogyo210 and @csidyel!

**Pull request closed without merging**